src/foam/dao/CSVSink.js

```diff
@@ -76,34 +101,97 @@ foam.CLASS({
             Null: function(value) {}
           }, function(value) {
             this.output_(value.toString());
-          })
+        }),
+      javaCode: `
+        StringBuilder sb = new StringBuilder();
+        sb.append(getCsv());
+        String s = value.toString();
+
+        if (s.indexOf("\\"") != -1 || s.indexOf(",") != -1) {
+          sb.append("\\"");
+          sb.append(s);
+          sb.append("\\"");
+        }
+        else {
+          sb.append(s);
+        }
+
+        setCsv(sb.toString());
+
+        sb.setLength(0);
+      `
     },
+    {
+      name: 'newLine_',
+      code: function() {
+        this.csv += '\n';
+        this.isNewLine = true;
+      },
+      javaCode: `
+        StringBuilder sb = new StringBuilder();
+
+        sb.append(getCsv());
+        sb.append("\\n");
+
+        setCsv(sb.toString());
 
-    function newLine_() {
-      this.csv += '\n';
-      this.isNewLine = true;
+        sb.setLength(0);
+        setIsNewLine(true);
+      `
     },
+    {
+      name: 'put',
+      code: function(obj) {
+        if ( ! this.of ) this.of = obj.cls_;
 
-    function put(obj) {
-      if ( ! this.of ) this.of = obj.cls_;
+        if ( ! this.isHeadersOutput ) {
+          this.props.forEach((element) => {
```

Comment:
Did you break this? Aren't props a StringArray now?

Reply:
:( ya i did. Fixed with next push